File: OpenMP_part2/task_array.c
#include "omp.h"
#include <stdio.h>
#include <malloc.h>
int main (int argc, char *argv[])
{
    int i=0;
    int n=20;

    //int array[n];
    int *array =(int*) malloc(sizeof(int)*n);
    // Init
    for (i=0;i<n;i++) array[i]=0;

    //Use openmp task
    i=0; 
    while (i<n)
    {
       array[i]=i;

       i++;

    }

    // Display array   
    for (i=0;i<n;i++) printf("array %d\n",array[i]);

}

File: OpenMP_part2/task_array.f90
program task_array
    implicit none
    integer, parameter :: n=20
    integer :: i
    integer, dimension (n) :: array

    ! Init
    do i=1,n
       array(i)=0
    end do

    !Use openmp task
    !$omp parallel
    !$omp task
    i=1
    do while (i<=n)

       array(i)=i

       i=i+1

    enddo
    !$omp end task


    !$omp single
    !$omp taskwait
    ! Display array
    do i=1,n
        write(*,*) array(i)
    enddo

    !$omp end single
    !$omp end parallel


end program task_array
